### ISSUE TITLE
Can't shift from an empty datastructure 

### ISSUE THREAD

**Issue by yuklia** [WyriHaximus/react-guzzle-psr7#44]
Hello! I've used your tool for making async requests to AWS S3 buckets. But i got an error:
```
In FutureTickQueue.php line 46:                                           
Can't shift from an empty datastructure 
```
It is weird because i don't need ticking manually event loop or calling wait as far as i use custom HTTP handler. 
```
        $loop = EventLoopFactory::create();
        $handler = new \WyriHaximus\React\GuzzlePsr7\HttpClientAdapter($loop);

        $s3Client = new S3Client([
            'http_handler' => \GuzzleHttp\HandlerStack::create($handler),
            'version' => 'latest',
            'region' => getenv('AWS_REGION'),
            'validate' => false
        ]);

        $dir1 = dirname(dirname(__DIR__)) . "/data/temp-test-medium-1";
        mkdir($dir1);
        $s3Client->downloadBucketAsync($dir1, 'test-medium-1')->then(function(){
            echo "Done 1\n";
        });

        $dir2 = dirname(dirname(__DIR__)) . "/data/temp-test-medium-2";
        mkdir($dir2);
        $s3Client->downloadBucketAsync($dir2, 'test-medium-2')->then(function(){
            echo "Done 2\n";
        });

        $loop->run();
        echo "finished\n";
```
Thanks in advance!

**Comment by WyriHaximus** [comment]
Hey @yuklia do you have a full stack trace for me? It is possible that something in Guzzle triggers the hackish internal manual loop ticking which could cause this issue.

**Comment by amel007** [comment]
Hey @yuklia @WyriHaximus 

        In FutureTickQueue.php line 44:                      
        Can't shift from an empty datastructure 

Shortest way to repeat situation is below:

        $params = [
            'handler' => HandlerStack::create(new HttpClientAdapter($loop)),
        ];

       $client = new Client($params);

       $client->get('url');
       $client->get('url');
       $client->get('url');
       $client->get('url');

but if change QUEUE_TIMER_INTERVAL to 0.001 sometimes to 0.0001
https://github.com/WyriHaximus/react-guzzle-psr7/blob/d5bf48c18775d2736c6876e80b25659e4302c172/src/HttpClientAdapter.php#L19-L29

it's work

**Comment by WyriHaximus** [comment]
@amel007 hmm that isn't configurable right now, would you be up to creating a PR that lets you configure that interval?

**Comment by WyriHaximus** [comment]
This has been addressed here in #48 and released as [`2.1.2`](https://github.com/WyriHaximus/react-guzzle-psr7/releases/tag/2.1.2), there is also an upstream PR at https://github.com/reactphp/event-loop/pull/186 by @claudiosdc solving this issue upstream.
